### PR TITLE
take tsconfig path aliasing  in account

### PR DESCRIPTION
**Pull request by ocoka**
Implement detection of alias configuration for a project in tsconfig 
(partial implementation, with TODO for webpack)

### CODE REVIEW

**Comment by aminya** [comment]
I think atom-typescript is a better place for this feature. We can add support for aliases in general though.

**Comment by ocoka** [comment]
Thank's for you attention, maybe it' matter for someone !
For me, this is not an issue for now, since all people that I know  to worried about this issue, and I had migrated to VsCode

**Comment by aminya** [comment]
> Thank's for you attention, maybe it's matter for someone !
> This is not an issue for now, since all people that I know to worried about this issue, had migrated to VsCode

This out of the topic of this PR but based on the download numbers or telemetry Atom has 3M+ active users, and those who want to use other IDEs are requesting us to write this plugin in other IDEs for them. #212 #223.

File: lib/autocomplete-paths.js [hunk 1]
```diff
@@ -112,6 +112,30 @@ export default {
...
+            if (tsconf.compilerOptions && tsconf.compilerOptions.paths) {
+              this._provider.setAliases(tsconf.compilerOptions.paths);
+            }
+          } catch(e) {
+            console.error('autocomplete-paths: can\'t parse "tsconfig.json" for aliases. incorrect json')
+          }
+        }
+      })
+    }
+  },
```
Comment (aminya) [inline lib/autocomplete-paths.js:137]: Can this be abstracted away to make aliases general? I suggest making this an option in the scopes.
https://github.com/atom-community/autocomplete-paths/blob/master/lib/config/default-scopes.js#L32


```
aliasReplacer: (projDir) => { // handle alias replacing }
```

**Comment by lierdakil** [comment]
Uh... I'm not entirely sure, but I think tsserver has its own support for path autocompletion, at least in import statements. I have autocomplete-paths disabled and still have no issue with autocompletion in typescript imports. That said, I don't generally use aliases. But tsserver should take tsconfig aliasing into account. So... I'm not convinced this is required actually..?

**Pull request closed without merging**